### PR TITLE
Fix two-stage runtime error when empty proposal

### PR DESCRIPTION
**Pull request by hhaAndroid**
## Motivation
Fix two-stage runtime error when empty proposal

Resolve: https://github.com/open-mmlab/mmdetection/issues/5037

## Modification
All two-stage models deal with the case of empty proposal specially.

### CODE REVIEW

**Review by jshilong — COMMENTED** [review]
Add unitests for empty `rois` can be better

**Comment by hhaAndroid** [comment]
> Add unitests for empty `rois` can be better

done

**Review by ZwwWayne — APPROVED** [review]

File: mmdet/models/roi_heads/cascade_roi_head.py [hunk 1]
```diff
@@ -317,13 +335,14 @@ def simple_test(self, x, proposal_list, img_metas, rescale=False):
...
-                                                       img_metas[j])
-                    for j in range(num_imgs)
-                ])
+                refine_roi_list = []
+                for j in range(num_imgs):
+                    if rois[j].shape[0] > 0:
+                        bbox_label = cls_score[j][:, :-1].argmax(dim=1)
+                        refine_roi = self.bbox_head[i].regress_by_class(
+                            rois[j], bbox_label[j], bbox_pred[j], img_metas[j])
+                        refine_roi_list.append(refine_roi)
```
Comment (ZwwWayne) [inline mmdet/models/roi_heads/cascade_roi_head.py:344]: 1. It seems that rois.shape==0 is not handled, will that have influence?
2. refine_roi -> refined_rois

File: mmdet/models/roi_heads/cascade_roi_head.py [hunk 2]
```diff
@@ -317,13 +335,14 @@ def simple_test(self, x, proposal_list, img_metas, rescale=False):
...
-                                                       img_metas[j])
-                    for j in range(num_imgs)
-                ])
+                refine_rois_list = []
+                for j in range(num_imgs):
+                    if rois[j].shape[0] > 0:
+                        bbox_label = cls_score[j][:, :-1].argmax(dim=1)
+                        refined_rois = self.bbox_head[i].regress_by_class(
+                            rois[j], bbox_label[j], bbox_pred[j], img_metas[j])
+                        refine_rois_list.append(refined_rois)
```
Comment (ZwwWayne) [inline mmdet/models/roi_heads/cascade_roi_head.py:344]: refined_roi_list

**Pull request merged**